ext/pdo/pdo_stmt.c

```diff
@@ -1209,9 +1225,9 @@ PHP_METHOD(PDOStatement, fetchObject)
 	PHP_STMT_GET_OBJ;
 	PDO_STMT_CLEAR_ERR();
 
-	if (!pdo_stmt_verify_mode(stmt, how, 0)) {
-		RETURN_FALSE;
-	}
+	/* Use pdo_stmt_verify_mode() to set fetch mode for this specific statement */
+	/* This should NOT fail */
+	pdo_stmt_verify_mode(stmt, PDO_FETCH_CLASS, 0, false);
```

Comment:
Hm, not that I see this again, your previous code using ZEND_ASSERT was fine, as we don't expect it to have side-effects in the success case. Though you might want to just drop the call entirely.

Reply:
I'm not sure I can drop it because `zend_argument_value_error(mod_arg_num)` does touch the stmt struct and change values from what I understand

Reply:
Hm, I'm not seeing it, pdo_stmt_verify_mode() looks purely read-only to me.

Reply:
Indeed... must have been confused by another function, will drop.